dashboard/app/controllers/datasets_controller.rb

```diff
@@ -21,6 +21,15 @@ def show
 
   # POST /datasets/:dataset_name/
   def update
+    records, columns = @firebase.csv_as_table(params[:csv_data])
+    @firebase.delete_shared_table URI.escape(params[:dataset_name])
+    response = @firebase.upload_shared_table(URI.escape(params[:dataset_name]), records, columns)
+    data = {status: response.code}
+    if response.success?
+      data[:records] = records
+      data[:columns] = columns
+    end
+    render json: data
```

Comment:
we should do the same thing here that we did this morning and bundle the status into our server's response so it isn't misinterpreted on the client

```ruby
data = {}
if response.success?
  data[:records] = records
  data[:columns] = columns
end

render json: data, status: response.code
```